server/jetstream_api.go

```diff
@@ -608,6 +636,17 @@ var (
 	jsNoAccountErr         = &ApiError{Code: 503, Description: "account not found"}
 )
 
+// ApiErrorByKind used by the cli to provide `nats error JS01`
+func ApiErrorByKind(kind string) *ApiError {
+	for _, v := range ApiErrors {
+		if v.Kind == kind {
+			return v
```

Comment:
They are unique yes. 

My thinking is that when writing or reading:

```
if !acc.JetStreamEnabled() {
		resp.Error = jsNotEnabledErr
		resp.Error = ApiErrors[JSNotEnabledErr]
		s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
		return
}
```

is easier to parse while scanning code due to the `JSNotEnabledErr` token than:

```
if !acc.JetStreamEnabled() {
		resp.Error = jsNotEnabledErr
		resp.Error = ApiErrors[JS1]
		s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
		return
}
```

This is perhaps not the best example but we have cases where we have like 3 or more if statements quite densely returning different errors.  For me when scanning such code I look for log lines and error strings rather than code - this retains a better readability to my eye

Reply:
Also you get the benefit of IDE completing errors for you and you dont need to remember whats the code for an error, you see error constants.

Reply:
Here is an earlier stab I took at doing this - we have many many errors:

https://github.com/ripienaar/nats-server/blob/cdef3e8acb07565336efda3c89b2e51be696bb99/server/jetstream_api.go#L600

I dont want to remember Kind codes.

Reply:
I did not make myself clear, I was saying that we can have 2 maps, one keyed by the id like you have (say JSNotEnabledErr, etc), and another that will be keyed by the kind ("JS01", "JS02", etc..). As you can see from my code snippet, I use both maps.
The advantage of the 2nd map (keyed by kind) is that there is no longer "slow lookup". The range and checking for kind basically becomes:
```
func ApiErrorByKind(kind string) *ApiError {
    return ApiErrorsByKind[kind]
}
```

Reply:
Sure, I dont think we have a need in the server for looking up by kind really and I dont want to forget to maintain one map all the time.  Can you think of a reason in the server to find by kind?

The CLI will loop when needed but its so fast - few 100 keys max - its effectively instantaneous.  I'd just remove the lookupbykind - there's no use. 

CLI will need to loop anyway as I want to be able to search/list/etc the kinds and so forth

Reply:
Oh ok, I got confused, I thought that you would get this lookup in the server that the server could query, etc..

Reply:
Yeah, no I dont think we really need that in the server